logrotate.c

```diff
@@ -2720,14 +2729,12 @@ static int readState(const char *stateFilename)
         /* treat non-openable file as an empty file for allocateHash() */
         f_stat.st_size = 0;
 
-        /* no error if state just not exists */
-        if (errno != ENOENT) {
-            message(MESS_ERROR, "error opening state file %s: %s\n",
-                    stateFilename, strerror(errno));
+        /* the statefile shoud exist, lockState() is holding a lock */
```

Comment:
This is misleading.  Holding the lock does not anyhow prevent `stateFilename` from being unlinked.  It should rather say "lockState() already created an empty state file in case it did not exist initially".

Reply:
Improved comment